src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Search/SearchIndexerTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using Hl7.Fhir.Model;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Context;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using Microsoft.Health.Fhir.Core.Models;
+using Microsoft.Health.Fhir.Tests.Common;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Serialization;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.Core.UnitTests.Features.Search
+{
+    public class SearchIndexerTests : IClassFixture<SearchParameterFixtureData>
+    {
+        private readonly SearchParameterFixtureData _fixtureData;
+        private ISearchIndexer _indexer;
+
+        private JsonSerializerSettings _settings = new JsonSerializerSettings
+        {
+            Converters = new List<JsonConverter>
+            {
+                new StringEnumConverter(),
+            },
+            ContractResolver = new CamelCasePropertyNamesContractResolver(),
+        };
+
+        public SearchIndexerTests(SearchParameterFixtureData fixtureData)
+        {
+            _fixtureData = fixtureData;
+
+            _indexer = new SearchIndexer(
+                () => _fixtureData.SearchDefinitionManager,
+                SearchParameterFixtureData.Manager,
+                new LightweightReferenceToElementResolver(new ReferenceSearchValueParser(new FhirRequestContextAccessor()), ModelInfoProvider.Instance),
+                NullLogger<SearchIndexer>.Instance);
+        }
+
+        [Theory]
+        [InlineData("DocumentReference-example")]
+        [InlineData("DocumentReference-example-002")]
+        [InlineData("DocumentReference-example-003")]
```

Comment:
Nit: it might make sense to indicate the different things that are getting tested in the test file names.

I don't know much about this, but, looking at the files, maybe something like this?
`DocumentReference-example-relatesTo-code-appends`
`DocumentReference-example-relatesTo-code-transforms-add-target`
`DocumentReference-example-relatesTo-code-transforms`

Reply:
Probably a good idea, these are the files used in the composite search tests, which I find the hardest to debug